lib/src/embedded/compiler.test.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import {Compiler} from './compiler';
+import {InboundMessage} from '../vendor/embedded_sass_pb';
+
+describe('compiler', () => {
+  let compiler: Compiler;
+  beforeEach(() => (compiler = new Compiler()));
+  afterEach(() => compiler.close());
+
+  it('compiles valid css input', async done => {
+    const input = new InboundMessage.CompileRequest.StringInput();
+    input.setSource('a {b: c}');
+    const request = new InboundMessage.CompileRequest();
+    request.setString(input);
+    const message = new InboundMessage();
+    message.setCompilerequest(request);
+
+    compiler.write$.next(message);
+    compiler.read$.subscribe(message => {
+      expect(message.getCompileresponse()?.getSuccess()?.getCss()).toEqual(
+        'a {\n  b: c;\n}'
+      );
+      done();
+    });
+  });
+
+  it('reports a ProtocolError', async done => {
+    compiler.write$.next(new InboundMessage());
+    compiler.error$.subscribe(error => {
+      expect(error.message).toEqual(
+        'Compiler reported error: InboundMessage.message is not set.'
+      );
+      done();
+    });
+  });
+
+  it('shuts itself down when a ProtocolError occurs', async done => {
+    compiler.write$.next(new InboundMessage());
+    compiler.read$.subscribe(
+      // Response callback
+      () => {},
```

Comment:
Is there an easy way to verify that the response and error callbacks are never called?

Reply:
I looked pretty seriously into marble testing for rxjs, but it doesn't work since the compiler's `read$` observable completes in a `process.nextTick()`, and there's no way for the marble test to execute after the tick. It's a shame, bc marble testing would be good for this kind of test.

Instead, I made the response and error branches manually fail the jasmine test.